codelab/JinghangCodeLab/MagicNumberConverterTest.java

```diff
@@ -0,0 +1,29 @@
+package JinghangCodeLab;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+public class MagicNumberConverterTest {
+    @Test
+    public void testConvertBinaryNumber() {
+        MagicNumberConverter convert = new MagicNumberConverter();
+        assertEquals(0, convert.binaryToDecimal("11111111111111111111111111111111111111111111111111"));
```

Comment:
create a separate test method for this sceanrio, "testNumberOutOfRange". you can use the following to verify that an exception is thrown for too large number:

`assertThrows(OutOfRangeException.class, convert.binaryToDecimal("11111111111111111111111111111111111111111111111"))`